src/hotspot/share/services/threadService.cpp

```diff
@@ -1278,6 +1278,7 @@ class GetThreadSnapshotHandshakeClosure: public HandshakeClosure {
       if (is_virtual) {
         // mounted vthread, use carrier thread state
         oop carrier_thread = java_lang_VirtualThread::carrier_thread(_thread_h());
+        assert(carrier_thread != nullptr, "should only get here for a mounted vthread");
```

Comment:
This assert will trigger if unmounted when the handshake op runs. This is one of the transition cases. In the loom this is now changed to return so that the caller retries.